config/config.go

```diff
@@ -927,8 +928,18 @@ func getDatabaseConfig(v *viper.Viper, networkID uint32) (node.DatabaseConfig, e
 		}
 	}
 
+	var (
+		dbName   = v.GetString(DBTypeKey)
+		readOnly = v.GetBool(DBReadOnlyKey)
+	)
+
+	if readOnly && dbName == memdb.Name {
+		return node.DatabaseConfig{}, fmt.Errorf("database type %s cannot be read only", dbName)
```

Comment:
Why do we prevent `memdb` from being read-only? I understand that is it is slightly redundant (because the memdb never writes anyways)... But it doesn't seem like the node would get into a bad situation... I feel like we can support both these flag options being passed concurrently and behave in an expected way

Reply:
Fair enough. Removed this check.